File: leetcode/75 leetcode/linked_list/reverse_linked_list/s.c
#include <stdlib.h>

// Definition for singly-linked list->
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode* hi;
    if (!head) return hi;
    hi->val = head->val;
    hi->next = NULL;
    while (!head->next) {
        head = head->next;
        struct ListNode* hey;
        hey->val = head->val;
        hey->next = head;
    }
    return hi;
}

File: leetcode/75 leetcode/linked_list/reverse_linked_list/s.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None:return 
        res = [ListNode(head.val, next=None)]
        while head.next != None:
            head = head.next
            res.insert(0, ListNode(val=head.val, next=res[0]))
        return res[0]
        